src/Tests/dotnet-new.Tests/TemplateDiscoveryTests.cs

```diff
@@ -0,0 +1,258 @@
+#pragma warning restore xUnit1004 // Test methods should not be skipped
+        public void CanReadCliDataFromDiff()
+        {
+            string testDir = CreateTemporaryFolder();
+            string packageLocation = PackTestNuGetPackage(_log);
+
+            new DotnetCommand(
+                _log,
+                "Microsoft.TemplateSearch.TemplateDiscovery.dll",
+                "--basePath",
+                testDir,
+                "--packagesPath",
+                Path.GetDirectoryName(packageLocation) ?? throw new Exception("Couldn't get package location directory"),
+                "-v",
+                "--diff",
+                "false")
+                .Execute()
+                .Should()
+                .ExitWith(0);
+
+            string[] cacheFilePaths = new[]
+            {
+                Path.Combine(testDir, "SearchCache", "NuGetTemplateSearchInfo.json"),
+                Path.Combine(testDir, "SearchCache", "NuGetTemplateSearchInfoVer2.json")
+            };
+            var settingsPath = CreateTemporaryFolder();
+            CheckTemplateOptionsSearch(cacheFilePaths, settingsPath);
+
+            string testDir2 = CreateTemporaryFolder();
+            new DotnetCommand(
+                _log,
+                "Microsoft.TemplateSearch.TemplateDiscovery.dll",
+                "--basePath",
+                testDir2,
+                "--packagesPath",
+                Path.GetDirectoryName(packageLocation) ?? throw new Exception("Couldn't get package location directory"),
+                "-v",
+                "--diff",
+                "true",
+                "--diff-override-cache",
+                Path.Combine(testDir, "SearchCache", "NuGetTemplateSearchInfoVer2.json"))
+                .Execute()
+                .Should()
+                .ExitWith(0)
+                .And.HaveStdOutContaining("not changed: 1");
+
+            string[] updatedCacheFilePaths = new[]
+            {
+                Path.Combine(testDir, "SearchCache", "NuGetTemplateSearchInfo.json"),
+                Path.Combine(testDir, "SearchCache", "NuGetTemplateSearchInfoVer2.json")
+            };
+            CheckTemplateOptionsSearch(updatedCacheFilePaths, settingsPath);
+        }
+
+        private void CheckTemplateOptionsSearch(IEnumerable<string> cacheFilePaths, string settingsPath)
+        {
+            foreach (var cacheFilePath in cacheFilePaths)
+            {
+                Assert.True(File.Exists(cacheFilePath));
+                new DotnetNewCommand(_log)
+                      .WithCustomHive(settingsPath)
+                      .WithEnvironmentVariable("DOTNET_NEW_SEARCH_FILE_OVERRIDE", cacheFilePath)
+                      .WithEnvironmentVariable("DOTNET_CLI_CONTEXT_VERBOSE", "true")
+                      .Execute()
+                      .Should()
+                      .ExitWith(0)
+                      .And.NotHaveStdErr();
+
+                new DotnetNewCommand(_log, "CliHostFile", "--search")
```

Comment:
```suggestion
                new DotnetNewCommand(_log, "CliHostFile", "search")
```

please change other usages to new syntax where possible